test/util/bitcoin-util-test.py

```diff
@@ -20,40 +20,44 @@
 import subprocess
 import sys
 
+
 def main():
     config = configparser.ConfigParser()
-    config.read_file(open(os.path.dirname(__file__) + "/../config.ini"))
+    config.read_file(open(os.path.join(os.path.dirname(__file__), "../config.ini")))
 
     parser = argparse.ArgumentParser(description=__doc__)
-    parser.add_argument('-v', '--verbose', action='store_true')
+    parser.add_argument("-v", "--verbose", action="store_true")
     args = parser.parse_args()
     verbose = args.verbose
 
     if verbose:
         level = logging.DEBUG
     else:
         level = logging.ERROR
-    formatter = '%(asctime)s - %(levelname)s - %(message)s'
+    formatter = "%(asctime)s - %(levelname)s - %(message)s"
     # Add the format/level to the logger
     logging.basicConfig(format=formatter, level=level)
 
-    bctester(config["environment"]["SRCDIR"] + "/test/util/data", "bitcoin-util-test.json", config["environment"])
+    bctester(os.path.join(config["environment"]["SRCDIR"], "test/util/data"), "bitcoin-util-test.json",
+             config["environment"])
+
 
-def bctester(testDir, input_basename, buildenv):
+def bctester(test_dir, input_basename, build_env):
     """ Loads and parses the input file, runs all tests and reports results"""
-    input_filename = testDir + "/" + input_basename
+    input_filename = os.path.join(test_dir, input_basename)
     raw_data = open(input_filename).read()
     input_data = json.loads(raw_data)
 
     failed_testcases = []
 
-    for testObj in input_data:
+    for test_obj in input_data:
         try:
-            bctest(testDir, testObj, buildenv)
-            logging.info("PASSED: " + testObj["description"])
-        except:
-            logging.info("FAILED: " + testObj["description"])
-            failed_testcases.append(testObj["description"])
+            bctest(test_dir, test_obj, build_env)
+            logging.info("PASSED: " + test_obj["description"])
+        except Exception as e:
+            logging.info("FAILED: " + test_obj["description"])
+            logging.error("Error %s: %s" % (test_obj["description"], e))
```

Comment:
Can you explain why it makes sense to first log to info and then the same to the error level?